tests/pin_project.rs

```diff
@@ -238,3 +238,27 @@ fn combine() {
     #[allow(unsafe_code)]
     unsafe impl<T: Unpin> UnsafeUnpin for Foo<T> {}
 }
+
+#[test]
+// This 'allow' is unrelated to the code
+// generated by pin-project - it's just to
+// allow us to put a private enum in a public enum
+#[allow(private_in_public)]
+fn test_private_type_in_public_type() {
+    #[pin_project]
+    pub struct PublicStruct<T> {
```

Comment:
My concern is that this generates the following documents (due to https://github.com/rust-lang/rust/issues/63281):
```rust
impl<T> Unpin for PublicStruct<T>
where
    __UnpinStructPublicStruct<T>: Unpin, 
```

Reply:
Even if `UnsafeUnpin` is used, a similar document is generated, but `UnsafeUnpin impl` is also documented at the same time, so we can know which type needs to be `Unpin`.
```rust
impl<T> Unpin for Foo<T>
where
    Wrapper<Self>: UnsafeUnpin,
```